RFCs/FS-1066-tryExactlyOne.md

```diff
@@ -0,0 +1,91 @@
+# F# RFC FS-1066 - Add tryExactlyOne to array, list and seq
+
+The design suggestion [Add tryExactlyOne to array, list and seq](https://github.com/fsharp/fslang-suggestions/issues/137) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+
+* [x] Approved in principle
+* [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/137)
+* [x] Details: [under discussion](https://github.com/fsharp/fslang-design/issues/344)
+* [x] Implementation: [In progress](https://github.com/Microsoft/visualfsharp/pull/5804)
+
+
+# Summary
+[summary]: #summary
+
+Add function `tryExactlyOne` working like `exactlyOne`, but notthrowing and returning `option` type to `array`, `list` and `seq` modules.
+
+# Motivation
+[motivation]: #motivation
+
+We want to increase the consistency of `FSharp.Core`. `tryExactlyOne` function is the missing safe (nonthrowing) deconstructor for `singleton` function and the safe alternative for `exactlyOne`. We expect to see users replacing their ad hoc implementations of `tryExactlyOne` and using it for their new use cases.
+
+# Detailed design
+[design]: #detailed-design
+
+The functions should copy the behavior of `exactlyOne`, but instead of throwing they should return `None`. The null check and `ArgumentNullException` throw for `array` and `seq` should not be replaced by `None` as calling the function with `null` is considered exceptional and undesired. 
+
+Example code:
+
+```fsharp
+List.tryExactlyOne []
+// None
+List.tryExactlyOne [1]
+// Some 1
+List.tryExactlyOne [1; 2]
+// None
+
+Array.tryExactlyOne null
+// ArgumentNullException
+Array.tryExactlyOne [||]
+// None
+Array.tryExactlyOne [|1|]
+// Some 1
+Array.tryExactlyOne [|1; 2|]
+// None
+
+Seq.tryExactlyOne null
+// ArgumentNullException
+Seq.tryExactlyOne (Seq.ofList [])
+// None
+Seq.tryExactlyOne (Seq.ofList [1])
+// Some 1
+Seq.tryExactlyOne (Seq.ofList [1; 2])
+// None
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Extending api surface.
```

Comment:
I'd also add that because this has different behavior from the LINQ equivalent, C# developers learning expecting something to throw may be a bit confused.

(Though I think most people will agree that returning `None` is more idiomatic)

Reply:
Added it.